clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2037,14 +2061,16 @@ class SyclKernelFieldChecker : public SyclKernelFieldHandler {
   }
 
   bool enterStruct(const CXXRecordDecl *, ParmVarDecl *, QualType) final {
-    // TODO
-    unsupportedFreeFunctionParamType();
+    // TODO manipulate struct depth once special types are supported for free
+    // function kernels.
+    // ++StructFieldDepth;
```

Comment:
Can we at least diagnose cases that involve SYCL special types now?

Reply:
IIRC this is already diagnosed by calling `handleOtherType `during visitation.

Reply:
Actually maybe not. It is not obvious to me whether we will hit that code since we aren't decomposing yet.

Reply:
No, we didn't diagnose because we don't decompose yet. I added diagnosing.